tealium/core/TealiumModulesManager.swift

```diff
@@ -73,20 +74,29 @@ class TealiumModulesManager : NSObject {
             return
         }
         
-        // Convert convience title to dictionary payload
-
-        var dataDictionary: [String : Any] = [TealiumKey.event: title ,
-                                              TealiumKey.eventName: title ,
-                                              TealiumKey.eventType: type.description() ]
+        // Dereference incoming args
+        let newTitle = title
+        let newInfo = info
         
-        if let additionalData = data {
-            dataDictionary += additionalData
-        }
+        // Convert convenience title to dictionary payload
+        var dataDictionary: [String : Any] = [TealiumKey.event: newTitle ,
+                                              TealiumKey.eventName: newTitle ,
+                                              TealiumKey.eventType: type.description() ]
+        if let newData = data { dataDictionary += newData }
         
+        // Create a more convenient track object to pass data and callback info with.
         let newTrack = TealiumTrack(data: dataDictionary,
-                                    info: info,
+                                    info: newInfo,
                                     completion: completion)
 
+        // Modules are still spinning up, delay track call
+        if self.allModulesReady() == false {
+            DispatchQueue.main.async {
+                self.track(newTrack)
+            }
+            return
```

Comment:
Not following.  Every time this is hit it will move the track call to the end of any existing requests. Should only ever occur if the library is still spinning up modules.